src/consensus/block_producer.rs

```diff
@@ -640,12 +690,43 @@ mod tests {
         disseminator
             .expect_send()
             .returning(|_| Box::pin(async { Ok(()) }));
-        let block_producer = setup(blockstore, pool, disseminator, Duration::from_micros(0));
+        let block_producer = setup(
+            blockstore.clone(),
+            pool,
+            disseminator,
+            Duration::from_micros(0),
+        );
+
+        let (parent_ready_tx, parent_ready_rx) = oneshot::channel();
+
+        let sk = &sks[epoch_info.leader(slot).id as usize];
+        let (_, _, shreds) = create_random_shredded_block(slot, 2, sk);
+        let mut parent_ready_tx = Some(parent_ready_tx);
```

Comment:
didn't know how else to get this to work with the borrow checker complaining about `parent_ready_tx` being moved in the loop